### DISCUSSION TITLE
More flexible way to exclude pages from the task list

### DISCUSSION THREAD

**Discussion by schiessle** [zim-desktop-wiki/zim-desktop-wiki#2322] [Ideas]
Hi all,

to follow the advice from the Issue template I discuss my idea here first.

The problem I'm trying to solve:

I use the task list plugin quite a lot. Every checkbox is a todo item. This is really handy because when I'm in a meeting and a todo comes up, I can just add a checkbox in the middle of the notes and it will show up in my task list. But sometimes I also use checkboxes where I don't want to create a task. One example is a preparation list for my next business trip. This checkboxes shouldn't show up in the task list.

Currently I can only exclude a path (and all sub pages). But this is to static. Using the business trip example my notes hierarchy looks something like this `Events:<random-event-name>:preparation` but there is also for example `Events:<random-event-name>:notes` which could contain tasks and many more sub pages. So I can't exclude everything under `Events:` and as `random-event-name` is dynamic (every time I organize another trip another event name comes up) I can't define a more detailed path to the `preparation` page.

The planing of a business trip is just a example. There are many more areas where this could make sense. Therefore I would like to have a more flexible way to exclude some pages from the task list. Some ideas:

1. Allow regular expressions, than I could define something like `Events:*:preparation`
2. Allow to define pages, independent from the path, than I could exlude just `preparation` no matter where it is located
3. Allow to exclude pages which contain a specific label, than I could define a label like `@notasks` and add it to the top of the relevant pages

What do you think? Would other benefit from such a additional option as well? Which solution would you prefer?

**Comment by jaap-karssenberg** [comment]
I'm open to patches that make the page selection for tasks more finegrained.

However, I would veto using an "@" tag for this - tags are meant to
organize content, not for page properties that are not content but
processing instructions.

Regards,

Jaap
